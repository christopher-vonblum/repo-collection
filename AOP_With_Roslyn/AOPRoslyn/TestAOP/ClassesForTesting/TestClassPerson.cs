using System;
using System.Collections.Generic;
using System.Text;

namespace TestAOP
{
    class TestClassPerson
    { 
        public TestClassPerson(string firstName, string lastName)
        {
            FirstName = firstName;
            LastName = lastName;
        }

        public string FirstName { get; }
        public string LastName { get; }

        public static TestClassPerson CopyConstructor(TestClassPerson t)
        {
            var newTC = new TestClassPerson(t.FirstName, t.LastName);
            return t;
        }
        public string Name(string separator = "")
        {
            return FirstName + separator + LastName;
        }
        
        public int Length()
        {
            return Name().Length;
        }
    }
}
