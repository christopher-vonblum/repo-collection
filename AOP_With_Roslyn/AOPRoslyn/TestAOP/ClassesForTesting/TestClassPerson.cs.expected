using System;
using System.Collections.Generic;
using System.Text;

namespace TestAOP
{
    class TestClassPerson
    { 
        public TestClassPerson(string firstName, string lastName)
        {
            FirstName = firstName;
            LastName = lastName;
        }

        public string FirstName { get; }
        public string LastName { get; }

        public static TestClassPerson CopyConstructor(TestClassPerson t)
{System.Console.WriteLine("start TestClassPerson_CopyConstructor_17");
#line 20
var newTC = new TestClassPerson(t.FirstName, t.LastName);
            return t;
System.Console.WriteLine("end TestClassPerson_CopyConstructor_17");}        public string Name(string separator = "")
{System.Console.WriteLine("start TestClassPerson_Name_22");
#line 25
return FirstName + separator + LastName;
System.Console.WriteLine("end TestClassPerson_Name_22");}        
        public int Length()
{System.Console.WriteLine("start TestClassPerson_Length_27");
#line 30
return Name().Length;
System.Console.WriteLine("end TestClassPerson_Length_27");}    }
}
